doc/operator-capabilities.md

```diff
@@ -0,0 +1,193 @@
+
+**Managed resources** - the Kubernetes objects or off-cluster services the Operator uses to constitute an Operand
+
+---
+
+## Level 1 - Basic Install
+
+The Operator offers the following basic features:
+
+**Installation of the workload**
+
+- Operator deploys an Operand or configures off-cluster resources
+- Operator waits for managed resources to reach a healthy state
+- Operator conveys readiness of application or managed resources to the user leveraging the `status` block of the Custom Resource
+
+**Example:** an Operator deploys a database by creating `Deployment`, `ServiceAccount`, `RoleBinding`, `ConfigMap`, `PersistentVolumeClaim` and `Secret` object, initializes an empty database schema and signals readiness of the database to accept queries.
+
+**Configuration of the workload**
+
+- Operator provides configuration via the `spec` section of the Custom Resource
+- Operator reconciles configuration and updates to it with the status of the managed resources
+
+**Example:** an Operator, managing a database, offers increasing the capacity of the database by resizing the underlying `PersistentVolumeClaim` based on changes the databases Custom Resource instance.
+
+**Guiding questions to determine Operator reaching Level 1**
+
+1. What installation configuration can be set in the CR?
+
+2. What additional CR installation configuration could still be added?
+
+3. Can you set operand configuration in the CR? If so, what configuration is supported for each operand?
+
+4. Does the managed application / workload get updated in a non-disruptive fashion when the configuration of the CR is changed?
+
+5. Does the status of the CR reflect that configuration changes are currently applied?
+
+6. What additional operand configuration could still be added?
+
+7. Do all of the instantiated CRs include a status block? If so, does it provide enough insight to the user about the application state?
+
+8. Do all of your CRs have documentation listing valid values and mandatory fields?
+
+---
+
+## Level 2 - Seamless Upgrades
+
+The Operator offers the following features related to upgrades:
+
+**Upgrade of the managed workload**
+
+- Operand can be upgraded in the process of upgrading the Operator, or
+- Operand can be upgraded as part of changing the CR
+- Operator understands how to upgrade older versions of the Operand, managed previously by an older version of the Operato
+
+**Upgrade of the Operator**
+
+- Operator can upgraded seamlessly and can either still manage older versions of the Operand or update them
+- Operator conveys inability to manage an unsupported version of the Operand in the `status` section of the CR
+
+**Example:** An Operator managing a database can update an existing database from a previous to a newer version without data loss. The Operator might do so as part of a configuration change or as part of an update of the Operator itself.
+
```

Comment:
I understand that we recommend users store the versions of the images that would be supported/use by the project in the Operator env vars, so an upgrade of its image would mean an upgrade of the Operator project. In this way, I think would be good add this info at least as a note here.

Reply:
Actually I think this deserves its own documentation in the SDK. Currently we do not tell people that they should not use inline references to images in their Operator source code but should rather inject this information via environment variables. The only reference is a [design proposal](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/contributors/design-proposals/related-images.md) in the OLM repo. 
I think a small note here will cause more confusion. Do you mind creating this doc?